.pipelines/templates/log-template.yaml

```diff
@@ -76,6 +76,26 @@ steps:
     displayName: Deployment Images
     condition: always()
 
+  - ${{ if eq(parameters.logType, 'failure') }}:
+    - bash: |
+        kubectl get pods -n kube-system -o custom-columns=NAME:.metadata.name,STATUS:.status.phase,NODE:.spec.nodeName
+        podList=`kubectl get pods -n kube-system -o custom-columns=NAME:.metadata.name,STATUS:.status.phase,NODE:.spec.nodeName --no-headers | grep -v Running | awk '{print $1}'`
+        array=($podList)
+
+        if [ -z ${array[0]} ]; then
+          echo "There are no kube-system pods in a non-ready state."
+        else
+          mkdir -p $acnLogs/${{ parameters.os }}non-ready
+          echo "Directory created: $acnLogs/${{ parameters.os }}non-ready"
+          echo "Capturing failed pods"
+          for pod in $podList; do
+            kubectl describe pod -n kube-system $pod > $acnLogs/${{ parameters.os }}non-ready/$pod.txt
+            echo "$acnLogs/${{ parameters.os }}non-ready/$pod.txt"
+          done
+        fi
+      displayName: Failure Logs
+      condition: always()
+
```

Comment:
because it was relocated and the diff shows the whole block as new, i'm struggling to see what has actually changed here

Reply:
The change allows us to catch if the privileged pod failed to reach ready state from any of the various ways we apply the daemonset to the cluster. 

Originally we would use kubectl to apply the daemonset again to retrieve logs from the node. In doing this we deleted the error logs and did not catch that it was the cause for the failure.